src/polydb.jl

```diff
@@ -0,0 +1,301 @@
+   return Database(client["polydb"])
+end
+
+# returns a Polymake.Polydb.Collection instance with the given name
+# sections and collections in the name are connected with the '.' sign,
+# i.e. names = "Polytopes.Lattice.SmoothReflexive"
+function get_collection(db::Database, name::String)
+   return Collection{Polymake.BigObject}(db.mdb[name])
+end
+
+function get_collection(db::Database, name::String, ::Type{T}) where T <: Union{Polymake.BigObject, Mongoc.BSON}
+   return Collection{T}(db.mdb[name])
+end
+
+# search a collection for documents matching the criteria given by d
+function find(c::Collection{T}, d::Dict=Dict(); opts::Union{Nothing, Dict}=nothing) where T
+   return Cursor{T}(Mongoc.find(c.mcol, Mongoc.BSON(d); options=opts))
+end
+
+function find(c::Collection{T}, d::Pair...) where T
+   return Cursor{T}(Mongoc.find(c.mcol, Mongoc.BSON(d...)))
+end
+
+# creating `BSON` iterators from the respective `Polymake.BigObject` iterator
+function Collection{Mongoc.BSON}(c::Collection{Polymake.BigObject})
+   return Collection{Mongoc.BSON}(c.mcol)
+end
+
+function Cursor{Mongoc.BSON}(cursor::Cursor{Polymake.BigObject})
+   return Cursor{Mongoc.BSON}(cursor.mcursor)
+end
+
+# returns a Polymake.BigObject from a Mongoc.BSON document
+function parse_document(bson::Mongoc.BSON)
+   str = Mongoc.as_json(bson)
+   return call_function(:common, :deserialize_json_string, str)
+end
+
+#Iterator
+
+Base.IteratorSize(::Type{<:Cursor}) = Base.SizeUnknown()
+Base.eltype(::Cursor{T}) where T = T
+
+# help functions to reduce runtime and keep code clean
+function _evaluateIteration(nothing::Nothing)
+   return nothing
+end
+
+function _evaluateIteration(a::Tuple{Mongoc.BSON, Any})
+   return (parse_document(a[1]), a[2])
+end
+
+# default iteration functions returning `Polymake.BigObject`s
+function Base.iterate(cursor::Cursor{Polymake.BigObject})
+   return _evaluateIteration(iterate(cursor.mcursor))
+end
+
+function Base.iterate(cursor::Cursor{Polymake.BigObject}, state::Nothing)
+   return _evaluateIteration(iterate(cursor.mcursor, state))
+end
+
+function Base.iterate(coll::Collection{Polymake.BigObject})
+   return _evaluateIteration(iterate(coll.mcol))
+end
+
+function Base.iterate(coll::Collection{Polymake.BigObject}, state::Mongoc.Cursor)
+   return _evaluateIteration(iterate(coll.mcol, state))
+end
+
+# functions for `BSON` iteration
+function Base.iterate(cursor::Cursor{Mongoc.BSON})
+   return iterate(cursor.mcursor)
+end
+
+function Base.iterate(cursor::Cursor{Mongoc.BSON}, state::Nothing)
+   return iterate(cursor.mcursor, state)
+end
+
+function Base.collect(cursor::Cursor{Mongoc.BSON})
+   return collect(cursor.mcursor)
+end
+
+function Base.iterate(coll::Collection{Mongoc.BSON})
+   return iterate(coll.mcol)
+end
+
+function Base.iterate(coll::Collection{Mongoc.BSON}, state::Mongoc.Cursor)
+   return iterate(coll.mcol, state)
+end
+
+#Info
+
+# prints a list of the fields of a collection
```

Comment:
it doesn't print anymore ;)

Reply:
what exactly do you want to copy in the iteration?
calling the `_evaluateIteration` helper not only sorts out the irregular cursor iteration, inside the documents are parsed to `Polymake.BigObject`s, too. iteration over a `Collection{Polymake.BigObject}` thus also already utilizes the canonical `iterate(x, ::Nothing) = nothing` and otherwise calls `_evaluateIteration(a::Tuple{Mongoc.BSON, Any})` (where parsing happens)